src/PathMappingRepository.php

```diff
@@ -107,7 +110,11 @@ public function listChildren($path)
      */
     public function hasChildren($path)
     {
-        return $this->getChildren($this->get($path), false)->count() !== 0;
+        if (!$this->resolveResource($path)) {
```

Comment:
Or maybe a better solution is to extract `getFilesystemPath($path)` out of `resolveResource()` and inline `getFirstFilesystemPath()` there. Then you can use the more efficient `getFilesystemPath()` where you do boolean checks and only create resources if they are really needed. You can then remove `resolveResource()` and use this instead:

```php
$resource = $this->createResource($this->getFilesystemPath($path), $path);
```

I would rename `createAndAttachResource()` to simply `createResource()` for brevity.